packages/api/src/router/absence.ts

```diff
@@ -0,0 +1,53 @@
+import { z } from "zod";
+
+import { adminProcedure, createTRPCRouter, protectedProcedure } from "../trpc";
+
+export const absenceRouter = createTRPCRouter({
+  createAbsence: protectedProcedure
+    .input(
+      z.object({
+        userId: z.number().nonnegative(),
+        absenceDate: z.date(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      await ctx.db.absence.create({
+        data: {
+          userId: input.userId,
+          absenceDate: input.absenceDate,
+        },
+      });
+    }),
+
+  getAllAbsences: adminProcedure.query(async ({ ctx }) => {
+    return await ctx.db.absence.findMany();
+  }),
+
+  getAbsenceByUserId: adminProcedure
+    .input(
+      z.object({
+        userId: z.number().nonnegative(),
+      }),
+    )
+    .query(async ({ input, ctx }) => {
+      return await ctx.db.absence.findMany({
+        where: {
+          userId: input.userId,
+        },
+      });
```

Comment:
sorry if previous comment was a bit unclear, but can you make this getAbsences. It would be a protetected procedure, and used by a parent to see all marked absences. It will take the user id for now, but in the future we will get  the user id  from our authentication

Reply:
no worries! thanks for the inputs, let me know if there's anything else to fix :)